### PR TITLE
fix: make error box dismissable

### PR DESCRIPTION
**Pull request by vipin8169**

### CODE REVIEW

**Review by coderbyheart — CHANGES_REQUESTED** [review]
This does not make the error box dismissable, instead it hides the error after 5 seconds.

This is a distinctly different solution, and has usability issues. If the user does not pay attention, or is on another screen while the error happens, they might not even see the error message.

This makes it also very hard for users to share error messages (because they will immediately disappear).

I have updated the issue description to reflect that user interaction should be needed to dismiss the error box.

**Comment by vipin8169** [comment]
How would you like to get it dismissed? do you want a cross button to dismiss it? what user action should make it disappear?

**Comment by coderbyheart** [comment]
A `X` icon is a typical UI element which allows to dismiss error messages, so yes, that could be a solution.

**Review by coderbyheart — CHANGES_REQUESTED** [review]
image

The positioning of the `X` button does not handle error message which are longer.

**Comment by vipin8169** [comment]
I am not sure what you mean. I request you to please elaborate and provide more information, and also a way to reproduce the problem.

**Comment by vipin8169** [comment]
Should I rather place the x button at the bottom right corner of this red message box?

**Comment by coderbyheart** [comment]
> I am not sure what you mean. I request you to please elaborate and provide more information, and also a way to reproduce the problem.

Add some text to the error message to make it longer, then you will see what is shown in the screenshot above.

> Should I rather place the x button at the bottom right corner of this red message box?

No, like in the development warning, the button should be in the top right corner.

image

**Comment by vipin8169** [comment]
Can you please take a look at the new changes and merge this pr if everything is as expected.

**Comment by coderbyheart** [comment]
Yes, I first want to set up a way to preview PRs (#46), otherwise I always have to check them out locally.

**Comment by vipin8169** [comment]
ok. Let me know if you would want to me to look at anything else in the meantime.

**Review by coderbyheart — CHANGES_REQUESTED** [review]
image

The close buttons in the red and the yellow box do not align.

**Comment by vipin8169** [comment]
Updated the code to align the two.

**Review by coderbyheart — CHANGES_REQUESTED** [review]
image

Now the text does not align with the one in the yellow box.

**Comment by vipin8169** [comment]
fixed

**Comment by coderbyheart** [comment]
I think you forgot to push, the last commit is 83926e19e7d5fe78f27ff8f4cdc9a6143a8533aa which is from 22 hours ago.

**Comment by vipin8169** [comment]
Sorry, my bad. pushed it now

**Review by coderbyheart — CHANGES_REQUESTED** [review]
image

Not that the vertical space between the button and the top of the box is different to the yellow box.

image

Please pay attention to these differences.

You could solve this by create an `DismissableBox` component which has the same styles for positioning text and close buttons and re-use this in both cases.

**Comment by vipin8169** [comment]
fixed

**Review by coderbyheart — CHANGES_REQUESTED** [review]
image

**Comment by vipin8169** [comment]
fixed.

File: src/Chat/components/Error.tsx [hunk 1]
```diff
@@ -1,27 +1,50 @@
...
-			<small>
-				UI version: <code>{GLOBAL_VERSION}</code>
-			</small>
-		</Text>
-	</Header>
-)
+export const Error = ({ type, message }: ErrorInfo) => {
+	const [visible, setVisible] = useState(true)
+	if (!visible) return null
+	else
```
Comment (coderbyheart) [inline src/Chat/components/Error.tsx:30]:
```suggestion
```
This is not neccessary. The return in the previous line will end the function execution.

**Review by coderbyheart — APPROVED** [review]
Ok, it now looks the same. I think this can be further improved in another PR and re-use the [Notice](https://github.com/distributeaid/chat-ui/blob/saga/src/Chat/components/Notice.tsx) component.

**Comment by vipin8169** [comment]
Can we merge this one?

**Comment by coderbyheart** [comment]
Yes, it's approved!

**Comment by vipin8169** [comment]
but it has not been merged yet :/

**Comment by coderbyheart** [comment]
Can't you do that?

**Comment by vipin8169** [comment]
I am not authorized to merge this pull request.

**Pull request merged**